services/command-router-infinispan/src/main/java/org/eclipse/hono/commandrouter/infinispan/app/Application.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.commandrouter.infinispan.app;
+
+import org.eclipse.hono.commandrouter.app.AbstractApplication;
+import org.eclipse.hono.deviceconnection.common.DeviceConnectionInfo;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+/**
+ * The Quarkus based Command Router main application class (Infinispan variant).
+ */
+@ApplicationScoped
+public class Application extends AbstractApplication {
```

Comment:
The Application class in both the infinispan and redis modules are exactly the same.

I moved the vast majority of logic out to a common class (AbstractApplication) but the injection of `DeviceConnectionInfo` must be done in the -infinispan and -redis modules where the concrete implementation of the `DeviceConnectionInfo` is created... injecting that in the -base module doesn't work since there is no implmentation of `DeviceConnectionInfo` available in the -base module.

Any suggestion for cleaning this up even more if possible would be appreciated.